crates/sui-transactional-test-runner/src/programmable_transaction_test_parser/parser.rs

```diff
@@ -284,22 +282,41 @@ where
         }
     }
 
-    pub fn parse_type_args_opt(&mut self) -> Result<Option<Vec<ParsedType>>> {
+    pub fn split_ty_args_opt(&mut self) -> Result<Option<Vec<String>>> {
         if !matches!(self.inner().peek_tok(), Some(CommandToken::TypeArgString)) {
             return Ok(None);
         }
         let contents = self.inner().advance(CommandToken::TypeArgString)?;
-        let type_tokens: Vec<_> = TypeToken::tokenize(contents)?
-            .into_iter()
-            .filter(|(tok, _)| !tok.is_whitespace())
-            .collect();
-        let mut parser = Parser::new(type_tokens);
-        parser.advance(TypeToken::Lt)?;
-        let res = parser.parse_list(|p| p.parse_type(), TypeToken::Comma, TypeToken::Gt, true)?;
-        parser.advance(TypeToken::Gt)?;
-        if let Ok((_, contents)) = parser.advance_any() {
-            bail!("Expected end of token stream. Got: {}", contents)
+        let contents = contents
+            .strip_prefix('<')
+            .and_then(|x| x.strip_suffix('>'))
+            .ok_or_else(|| {
+                anyhow::anyhow!("Expected type argument string to start with '<' and end with '>'")
+            })?;
+
+        let mut res = vec![];
+        let mut open_pins = 0;
+        let mut start = 0;
+        for (i, c) in contents.char_indices() {
+            match c {
+                '<' => open_pins += 1,
+                '>' => open_pins -= 1,
+                ',' if open_pins == 0 => {
+                    res.push(contents[start..i].to_string());
+                    start = i + 1;
+                }
+                _ => {}
```

Comment:
Not sure what is going on here, but it feels like something we should do with a parser?

Reply:
All gone -- parsing it in directly now :)